src/UNIVesting.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.29;
+
+import {
+  BokkyPooBahsDateTimeLibrary as DateTime
+} from "BokkyPooBahsDateTimeLibrary/contracts/BokkyPooBahsDateTimeLibrary.sol";
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {IUNIVesting} from "./interfaces/IUNIVesting.sol";
+
+/// @title UNIVesting
+/// @notice A vesting contract that releases UNI tokens quarterly to a designated recipient
+/// @dev The contract unlocks its first tranche on Jan 1, 2026 and allows withdrawals every calendar
+/// quarter thereafter. The owner must maintain an ERC20 allowance for the contract to transfer UNI
+/// tokens. Supports partial withdrawals when allowance is less than vested amount.
+contract UNIVesting is Owned, IUNIVesting {
+  using SafeTransferLib for ERC20;
+
+  /// @notice Number of months in a quarter
+  uint256 private constant MONTHS_PER_QUARTER = 3;
+
+  /// @dev equivalent to January 1, 2026 00:00:00 UTC
+  uint256 private constant FIRST_UNLOCK_TIMESTAMP = 1_767_225_600;
+
+  /// @inheritdoc IUNIVesting
+  ERC20 public immutable UNI;
+
+  /// @inheritdoc IUNIVesting
+  uint256 public quarterlyVestingAmount = 5_000_000 ether;
+
+  /// @inheritdoc IUNIVesting
+  address public recipient;
+
+  /// @inheritdoc IUNIVesting
+  uint48 public lastUnlockTimestamp;
```

Comment:
nit: this could be named a bit more explicitly bc from my understanding this MUST lie exactly on a quarter